lib/plug/adapters/cowboy/conn.ex

```diff
@@ -69,7 +69,14 @@ defmodule Plug.Adapters.Cowboy.Conn do
     # otherwise cowboy will attempt to load the
     # whole length at once.
     {limit, opts} = Keyword.pop(opts, :length, 8_000_000)
-    {:ok, limit, acc, req} = parse_multipart(Request.part(req), limit, opts, [], callback)
+
+    # We need to construct the header opts using fix defaults here, since once opts
+    # are passed cowboy defaults are not applied anymore and fallbacks in body function are used.
+    header_opts = [length: 64000,
+                   read_length:  opts[:header_read_length]  || 64000,
+                   read_timeout: opts[:header_read_timeout] ||  5000 ]
+
+    {:ok, limit, acc, req} = parse_multipart(Request.part(req, header_opts), limit, opts, [], callback)
```

Comment:
As far as I can see, this is the only call of `Request.part` where the `header_opts` should be passed, right? https://github.com/elixir-lang/plug/blob/master/lib/plug/adapters/cowboy/conn.ex#L102